File: atv_academia/Recepcionista.java
import java.util.Date;
import java.util.HashSet;
import java.util.Set;
import java.util.StringJoiner;

public class Recepcionista extends Funcionario {
    private HashSet<Horario> horarios;
    public Recepcionista(String matricula, String nome) {
        super(matricula, nome);
        this.horarios = new HashSet<>();
    }

    public void addHorario(Horario horario) {
        this.horarios.add(horario);
    }

    public void removeHorario(Horario horario) {
        this.horarios.remove(horario);
    }

    public HashSet<Horario> getHorarios() {
        return horarios;
    }

    public String getTipoFuncionario(){
        return "imd.academia.Recepcionista";
    }

    public String getHorariosPrint(){
        StringJoiner horarios_str = new StringJoiner(", ");
        for (Horario horario : this.horarios) {
            horarios_str.add(horario.toString());
        }
        return String.format("{ %s }", horarios_str.toString());
    }
}

File: atv_academia_kotlin/src/main/kotlin/Recepcionista.kt
import java.util.*

class Recepcionista(matricula: String?, nome: String?) : Funcionario(matricula, nome) {
    val horarios: HashSet<Horario>

    init {
        horarios = HashSet()
    }

    fun addHorario(horario: Horario) {
        horarios.add(horario)
    }

    fun removeHorario(horario: Horario) {
        horarios.remove(horario)
    }

    override val tipoFuncionario: String
        get() = "imd.academia.Recepcionista"
    val horariosPrint: String
        get() {
            val horarios_str = StringJoiner(", ")
            for (horario in horarios) {
                horarios_str.add(horario.toString())
            }
            return "{ ${ horarios_str.toString() } }"
        }
}
